var/spack/repos/builtin/packages/py-torchgeo/package.py

```diff
@@ -35,113 +32,112 @@ class PyTorchgeo(PythonPackage):
-    depends_on("py-pytorch-lightning@1.5.1:1.8", when="@0.3.0", type=("build", "run"))
-    depends_on("py-pytorch-lightning@1.3:1.8", when="@:0.2", type=("build", "run"))
-    depends_on("py-rasterio@1.0.20:1", when="@0.3:", type=("build", "run"))
-    depends_on("py-rasterio@1.0.16:1", when="@:0.2", type=("build", "run"))
-    depends_on("py-rtree@1", when="@0.3:", type=("build", "run"))
-    depends_on("py-rtree@0.9.4:1", when="@0.2.1", type=("build", "run"))
-    depends_on("py-rtree@0.5:1", when="@:0.2.0", type=("build", "run"))
-    depends_on("py-scikit-learn@0.21:1", when="@0.3:", type=("build", "run"))
-    depends_on("py-scikit-learn@0.18:1", when="@:0.2", type=("build", "run"))
-    depends_on("py-segmentation-models-pytorch@0.2:0.3", when="@0.3.1:", type=("build", "run"))
-    depends_on("py-segmentation-models-pytorch@0.2", when="@:0.3.0", type=("build", "run"))
-    depends_on("py-shapely@1.3:2", when="@0.4:", type=("build", "run"))
-    depends_on("py-shapely@1.3:1", when="@:0.3", type=("build", "run"))
-    depends_on("py-timm@0.4.12:0.6", when="@0.4:", type=("build", "run"))
-    depends_on("py-timm@0.4.12:0.4", when="@:0.3", type=("build", "run"))
-    depends_on("py-torch@1.12:2", when="@0.4.1:", type=("build", "run"))
-    depends_on("py-torch@1.12:1", when="@0.4.0", type=("build", "run"))
-    depends_on("py-torch@1.9:1", when="@0.2:0.3", type=("build", "run"))
-    depends_on("py-torch@1.7:1", when="@0.1", type=("build", "run"))
-    depends_on("py-torchmetrics@0.10:0.11", when="@0.4:", type=("build", "run"))
-    depends_on("py-torchmetrics@0.7:0.9", when="@0.3", type=("build", "run"))
-    depends_on("py-torchmetrics@0.7:0.8", when="@0.2.1", type=("build", "run"))
-    depends_on("py-torchmetrics@0.7", when="@:0.2.0", type=("build", "run"))
-    depends_on("py-torchvision@0.13:0.15", when="@0.4.1:", type=("build", "run"))
-    depends_on("py-torchvision@0.13:0.14", when="@0.4.0", type=("build", "run"))
-    depends_on("py-torchvision@0.10:0.13", when="@0.3", type=("build", "run"))
-    depends_on("py-torchvision@0.10:0.12", when="@0.2", type=("build", "run"))
-    depends_on("py-torchvision@0.3:0.12", when="@0.1", type=("build", "run"))
+    depends_on("py-lightly@1.4.4:", when="@0.5:", type=("build", "run"))
+    depends_on("py-lightning@2:", when="@0.5:", type=("build", "run"))
+    depends_on("py-lightning@1.8:", when="@0.4.1:", type=("build", "run"))
+    depends_on("py-matplotlib@3.3.3:", when="@0.5:", type=("build", "run"))
+    depends_on("py-matplotlib@3.3:", type=("build", "run"))
+    depends_on("py-numpy@1.19.3:", when="@0.5:", type=("build", "run"))
+    depends_on("py-numpy@1.17.2:", type=("build", "run"))
+    depends_on("py-pandas@1.1.3:", when="@0.5:", type=("build", "run"))
+    depends_on("pil@8:", when="@0.5:", type=("build", "run"))
+    depends_on("pil@6.2:", type=("build", "run"))
+    depends_on("pil+jpeg+tiff+zlib", type=("build", "run"))
```

Comment:
Should we add comments here where these additional dependencies come from?
(same later in the `dataset` variant with the `gdal` and `libtiff` dependencies)

Reply:
Done. TL;DR: TorchGeo uses rasterio (GDAL/libtiff), PIL (pillow), and cv2 (OpenCV) for file I/O. We need to control how some of these indirect dependencies are built in order to ensure that we can open files from all of the datasets we currently support. We'll likely add to these in the future.